flink-clients/src/main/java/org/apache/flink/client/program/ContextEnvironment.java

```diff
@@ -87,9 +87,11 @@ public JobExecutionResult execute(String jobName) throws Exception {
 		ClientUtils.addJarFiles(jobGraph, this.jarFilesToAttach);
 		jobGraph.setClasspaths(this.classpathsToAttach);
 
-		lastJobExecutionResult = client
-				.submitJob(jobGraph, this.userCodeClassLoader)
-				.getJobExecutionResult();
+		if (detached) {
+			lastJobExecutionResult = new DetachedJobExecutionResult(ClientUtils.submitJob(client, jobGraph).getJobID());
```

Comment:
You do not need to re-wrap it in a `DetachedJobExecutionResult`. That is what the `ClientUtils.submitJob()` returns.

Reply:
Thanks. Change the return type of `ClientUtils.submitJob` to `JobExecutionResult` which doesn't break others.